ceph: Look for binaries on PATH if not found

krig opened the pull request:
**Description of your changes:**

Similarly to #2675, try to find the rook binary
on PATH in case it isn't found in /usr/local/bin/rook.

Signed-off-by: Kristoffer Grönlund <kgronlund@suse.com>

// known CI issues
[skip ci]

leseb commented:
@krig why would the binary location change? Have you seen a bug with this?

krig commented:
@leseb we are packaging rook for openSUSE/SLES, and packages putting binaries in `/usr/local` is frowned upon - it violates the FHS. So we tried installing them to `/usr` instead, but hit this hardcoded path which only looks in `/usr/local/bin/rook`. So this patch is meant to let things work even if rook has been installed as `/usr/bin/rook` instead.

leseb commented:
@krig are you saying that putting the `rook` binary in `/usr/local/bin` violates FHS? From what I can see FHS says: `Most of the time /usr/local is used for installing software/data that are not part of the standard operating system distribution`, so this looks like the right place to me.

krig commented:
@leseb Yeah, I think that sentence is misleading. The FHS, section 4.9.1 says

```
The /usr/local hierarchy is for use by the system administrator
   when installing software locally. It needs to be safe from
   being overwritten when the system software is updated.
```

So anything considered "system software" should not go into `/usr/local`, meaning anything updated by the distribution package system, it is reserved for use by the local administrator. In our case, rook would be considered part of the system software.

krig commented:
Here's what rpmlint says in our case:

```
rook-rookflex.x86_64: W: suse-filelist-forbidden-fhs23 /usr/local is not allowed in FHS 2.3
rook.x86_64: W: suse-filelist-forbidden-fhs23 /usr/local is not allowed in FHS 2.3
see http://www.pathname.com/fhs/ for a better location
```

leseb commented:
@krig thanks for the clarification, I missed the downstream package aspect, it makes perfect sense.

krig commented:
@BlaineEXE @leseb Thank you! Just a heads up, I simplified the patch slightly and added a note to the commit message clarifying the reason for the patch.

leseb commented:
@krig thanks, we should be able to merge soon.

BlaineEXE commented on pkg/daemon/ceph/osd/init.go:
```diff
@@ -89,6 +90,13 @@ func copyBinary(sourceDir, targetDir, filename string) error {
+	// Check if the source path exists, and look in PATH if it doesn't
+	if _, err := os.Stat(sourcePath); err != nil {
+		if sourcePath, err = exec.LookPath(filename); err != nil {
+			return nil
```
If the file isn't found in the path, this should return the error, not nil.

krig replied:
makes sense, I'll do that.
